File: app/src/main/java/me/zhanghai/android/files/ftpserver/FtpServerReceiver.java
/*
 * Copyright (c) 2019 Hai Zhang <dreaming.in.code.zh@gmail.com>
 * All Rights Reserved.
 */

package me.zhanghai.android.files.ftpserver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import java.util.Objects;

import androidx.annotation.NonNull;

public class FtpServerReceiver extends BroadcastReceiver {

    private static final String KEY_PREFIX = FtpServerReceiver.class.getName() + '.';

    public static final String ACTION_STOP = KEY_PREFIX + "STOP";

    @Override
    public void onReceive(@NonNull Context context, @NonNull Intent intent) {
        String action = intent.getAction();
        if (Objects.equals(action, ACTION_STOP)) {
            FtpServerService.stop(context);
        } else {
            throw new IllegalArgumentException(action);
        }
    }
}

File: app/src/main/java/me/zhanghai/android/files/ftpserver/FtpServerReceiver.kt
/*
 * Copyright (c) 2019 Hai Zhang <dreaming.in.code.zh@gmail.com>
 * All Rights Reserved.
 */

package me.zhanghai.android.files.ftpserver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import me.zhanghai.android.files.app.application
import me.zhanghai.android.files.app.clipboardManager
import me.zhanghai.android.files.util.copyText

class FtpServerReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        when (val action = intent.action) {
            ACTION_STOP -> FtpServerService.stop(context)
            ACTION_COPY_URL -> {
                val url = FtpServerUrl.getUrl()
                if (url != null) {
                    clipboardManager.copyText(url, context)
                }
            }
            else -> throw IllegalArgumentException(action)
        }
    }

    companion object {
        const val ACTION_STOP = "stop"
        const val ACTION_COPY_URL = "copy_url"

        fun createIntent(): Intent =
            Intent(application, FtpServerReceiver::class.java)
                .setAction(ACTION_STOP)
    }
}
